From d3fc3f94fe532257b224da8e080087e32a973dfc Mon Sep 17 00:00:00 2001
From: jzhuan5 <jin.can.zhuang@intel.com>
Date: Fri, 7 Dec 2012 18:19:45 -0500
Subject: [PORT FROM MAIN] usb:adbd: sleep 1 seconds before termination

BZ: 72801

Upon restarting adbd, it sends CLOSE command to remote and terminates
itself. But the CLOSE command may not be sent if adbd terminates
before input_thread finishing sending the command. Adding a delay of 1
second should greatly prevent this case from happening.

Change-Id: I861d407b8daf1c3a61bf779ceef5787735c1c9a4
Orig-Change-Id: Ife8e8ab49dd454d14b8a9756fc2f6d0a05620fb9
Signed-off-by: jzhuan5 <jin.can.zhuang@intel.com>
Reviewed-on: http://android.intel.com:8080/83160
Tested-by: Wang, Ran <ran.wang@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 adb/sockets.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/adb/sockets.c b/adb/sockets.c
index 5b56ed3..587bfa4 100644
--- a/adb/sockets.c
+++ b/adb/sockets.c
@@ -219,6 +219,7 @@ static void local_socket_destroy(asocket  *s)
 
     if (exit_on_close) {
         D("local_socket_destroy: exiting\n");
+        sleep(1);
         exit(1);
     }
 }
-- 
1.7.4.1

